helix-core/src/main/java/org/apache/helix/task/ThreadCountBasedTaskAssignmentCalculator.java

```diff
@@ -146,6 +147,20 @@ public ThreadCountBasedTaskAssignmentCalculator(TaskAssigner taskAssigner,
     return taskAssignment;
   }
 
+  @Override
+  public Set<Integer> getRemovedPartitions(JobConfig jobConfig, JobContext jobContext, Set<Integer> allPartitions) {
```

Comment:
Since this is same implementation as generic one, can we have this implementation as default implement in abstract class? Then if we need difference implementation, let override it.

Reply:
I think it is safe to keep it as it is because for other methods the same rule is applied. For example, getAllTaskPartiotions are the same in both classes.

Reply:
As we discussed offline, I will address this.